### ISSUE TITLE
Basic auto-indent support

### ISSUE THREAD

**Issue by MrYann** [Gimly/vscode-fortran#6]
Hi : )

As you surely know, VSCode just implemented an auto-indent feature. I tried to create a pull request, but since the old one is still open this gets a bit messy. So, I propose here is a very basic suggestion to append to the file "modern.configuration.json": 
```json
	"indentationRules": {
		"increaseIndentPattern": "^\\s*((module|contains|function|subroutine|begin|else|elsif|do|for|case|select case|type)|(if\\b.*\\bthen))\\b",
		"decreaseIndentPattern": "^\\s*(contains|end|else|elseif|endif|case)\\b"
	}
```

Note that unfortunately VSCode does not seem to correct indentation within pasted blocks, so you have to write your own lines to test the feature (after setting `editor.autoIndent` to true)

I use Fortran and regexes very scarcely, so my proposition will probably have to be enhanced.

**Comment by Gimly** [comment]
Did you check this tool? https://github.com/pseewald/fprettify

Seems to be interesting, but it's quite the work to reproduce.

**Comment by AmirulOm** [comment]
@Gimly , would it be possible to just use fprettify as the back engine for the auto indent in this plugin?. i tried fprettify and it was wonderfull.

**Comment by trolley813** [comment]
@Gimly Will it be possible to use `fprettify` as a code formatter (adding "Format Code" context menu action)? It does full formatting, as I know.

**Comment by Gimly** [comment]
I had looked at it, but the issue I see is that it's a Python tool. I can't therefore use it as a dependency for the fortran extension. I could use it as an external dependency and add a parameter so that provided a user has installed it, he can reference it and it will be used to format the code.

Is that the approach you suggest?

**Comment by trolley813** [comment]
@Gimly Yes, it is. Of course, having an optional dependency is a much more flexible approach.

**Comment by Gimly** [comment]
OK, then I'll try to start working on it as soon as possible, it shouldn't be too complicated hopefully.
